src/swift/Time.swift

```diff
@@ -119,6 +119,40 @@ public func ==(a: DispatchWallTime, b: DispatchWallTime) -> Bool {
 	return a.rawValue == b.rawValue
 }
 
+// Returns m1 * m2, clamped to the range [Int64.min, Int64.max].
+// Because of the way this function is used, we can always assume
+// that m2 > 0.
+private func clampedInt64Product(_ m1: Int64, _ m2: Int64) -> Int64 {
+	assert(m2 > 0, "multiplier must be positive")
+	let (result, overflow) = m1.multipliedReportingOverflow(by: m2)
+	guard overflow == false else {
```

Comment:
This reads a little backwards to me. What do you think about the following instead?

```swift
if overflow {
  return m1 > 0 ? Int64,max : Int64.min
}
return result
```

Reply:
Fixed.